vuepress/docs/next/tutorials/consume/external-db-ms.md

```diff
@@ -0,0 +1,203 @@
+                            "name": "ed1d60fc-285a480f-external-books-service-mysql-secret",
+                            "key": "SPRING_DATASOURCE_PASSWORD"
+                        }
+                    }
+                },
+                {
+                    "name":"SPRING_DATASOURCE_URL",
+                    "valueFrom": {
+                        "secretKeyRef": {
+                            "name": "ed1d60fc-285a480f-external-books-service-mysql-secret",
+                            "key": "SPRING_DATASOURCE_URL"
+                        }
+                    }
+                },
+                {
+                    "name":"SPRING_JPA_DATABASE_PLATFORM",
+                    "value": "org.hibernate.dialect.MySQL8Dialect"
+                }
+            ]
+        }
+    ]
+    ...
+    "name": "YOUR-BUNDLE-NAME",
+    "version": "1.0.0",
+    "description": "a description",
+    "type": "bundle"
+}
+```
+3. Pack and publish your bundle:
+```
+  ent bundle pack
+  ent bundle publish
+```
+
+### Create and Apply a Plugin Configuration Secret
+Prior to installing a bundle, a plugin configuration Secret must be created to provide a different datasource for each tenant. Create a Secret for each microservice in the bundle to provide a series of values required by the Spring Dynamic Datasource Routing for every tenant as needed.  
+
+1. Retrieve the bundle ID and the plugin code:
+ ```
+ ent ecr get-bundle-id docker://registry.hub.docker.com/YOUR-ORG/YOUR-BUNDLE
+ ```
+ ```
+ ent ecr get-plugin-code YOUR-ORG/YOUR-MS-NAME --repo=docker://registry.hub.docker.com/YOUR-ORG/YOUR-BUNDLE
+ ```
+ 
+2. Create the JSON plugin configuration Secret for the microservice plugin with a name that reflects the bundle ID and plugin code as Kuternetes is not tenant aware.
+
+   The name should begin with "pn" plus the concatenation of the bundle ID, plugin code, and microservice name, connected by dashes. The name can be appended with suffixes such as "SECRET" for better identification.  
+
+   E.g.  
+   Bundle ID = `hasldk12`, Plugin Code = `8dsjahj2`, Microservice name = YOUR-MS-NAME  
+   Your Configuration Secret= `pn-hasldk12-8dsjahj2-YOUR-MS-NAME-SECRET`
+
+3. Add the following env vars per tenant. The key for your primary tenant must be 'primary' regardless of what the actual tenant ID is. Secondary tenant keys should refer to their tenant IDs. Add the key-value pairs for the Spring Datasource Username, password, URL, and connection driver, in a base 64 encoded format.
+
+This is a decrypted example of a plugin configuration Secret:
+``` json
+apiVersion: v1
+kind: Secret
+metadata:
+  name: pn-hasldk12-8dsjahj2-YOUR-MS-NAME-SECRET
+  namespace: YOUR-NAMESPACE
+type: Opaque
+data:
+  primary: >-
+    {
+  "environment_variables": [
+    {
+      "name": "SPRING_DATASOURCE_USERNAME",
+      "valueFrom": {
+        "secretKeyRef": {
+          "name": "hasldk12-8dsjahj2-YOUR-SECRET-PRIMARY",
+          "key": "YOUR-USERNAME"
```

Comment:
This `key` (as the other ones in this section) is not a value. We are referencing other secrets that will contain our data. So in this case the key will be only a name used to access our desired value. So we want to take the username value by the secret named `hasldk12-8dsjahj2-YOUR-SECRET-PRIMARY`. Then within this referenced secret (that can contain a lot of key-value pairs) we want to access the value corresponding to the `username` key.

E.g.

The secret `hasldk12-8dsjahj2-YOUR-SECRET-PRIMARY` contains 3 key-value pairs:

- `username`: `my_username_23345`
- `password`: `ent4ndo_passwd`
- `url`: `jdbs://localhost:5432/mydatabase`

In the plugin configuration secret we are saying that an environment variable named `SPRING_DATASOURCE_USERNAME` must be injected in the MS and its value must be taken from the secret named `hasldk12-8dsjahj2-YOUR-SECRET-PRIMARY`, taking the value corresponding to the `username` key, so the value `my_username_23345`.
Similar thing for the other 2 env vars `SPRING_DATASOURCE_PASSWORD` and `SPRING_DATASOURCE_URL`.

The `SPRING_JPA_DATABASE_PLATFORM` instead specifies a direct value to inject in the env var (using `value` instead of `valueFrom`). This is because the database platform is not protected data, so we can inject it in the MS as plain text instead of referencing it from another secret